### PR TITLE
[people] Always search Bugzilla accounts using Mozilla emails

### LINKED ISSUE #1529: Employees with unlinked Bugzilla accounts
When an employee does not use their Mozilla email for Bugzilla and at the same time, their Bugzilla email is not linked in the phonebook (e.g., https://people.mozilla.org/p/masayuki-nakano). This causes some tools to fail:

```
2022-07-11 12:07:48,210 - ERROR - Tool no_severity
Traceback (most recent call last):
  File "/home/rm_bot/relman-auto-nag/auto_nag/bzcleaner.py", line 738, in run
    self.send_email(date=date)
  File "/home/rm_bot/relman-auto-nag/auto_nag/bzcleaner.py", line 671, in send_email
    title, body = self.get_email(date)
  File "/home/rm_bot/relman-auto-nag/auto_nag/bzcleaner.py", line 632, in get_email
    data = self.get_email_data(date, bug_ids)
  File "/home/rm_bot/relman-auto-nag/auto_nag/bzcleaner.py", line 623, in get_email_data
    bugs = self.get_bugs(date=date, bug_ids=bug_ids)
  File "/home/rm_bot/relman-auto-nag/auto_nag/bzcleaner.py", line 442, in get_bugs
    timeout=self.get_config("bz_query_timeout"),
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/libmozdata/bugzilla.py", line 176, in wait
    self.wait_bugs()
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/libmozdata/bugzilla.py", line 187, in wait_bugs
    r.result()
  File "/usr/local/lib/python3.7/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/local/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/requests/sessions.py", line 653, in send
    r = dispatch_hook('response', hooks, r, **kwargs)
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/requests/hooks.py", line 31, in dispatch_hook
    _hook_data = hook(hook_data, **kwargs)
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/libmozdata/bugzilla.py", line 552, in __bugs_cb
    self.bughandler.handle(bug)
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/libmozdata/handler.py", line 15, in handle
    self.handler(*args)
  File "/home/rm_bot/relman-auto-nag/auto_nag/bzcleaner.py", line 324, in bughandler
    auto_ni = self.get_mail_to_auto_ni(bug)
  File "/home/rm_bot/relman-auto-nag/auto_nag/scripts/workflow/no_severity.py", line 75, in get_mail_to_auto_ni
    mail, nick = self.round_robin.get(bug, self.date)
  File "/home/rm_bot/relman-auto-nag/auto_nag/round_robin.py", line 174, in get
    nick = self.get_nick(bzmail, pc, cal)
  File "/home/rm_bot/relman-auto-nag/auto_nag/round_robin.py", line 132, in get_nick
    BugzillaUser(user_names=[bzmail], user_handler=handler).wait()
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/libmozdata/connection.py", line 134, in wait
    r.result()
  File "/usr/local/lib/python3.7/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/local/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/requests/sessions.py", line 653, in send
    r = dispatch_hook('response', hooks, r, **kwargs)
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/requests/hooks.py", line 31, in dispatch_hook
    _hook_data = hook(hook_data, **kwargs)
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/libmozdata/connection.py", line 123, in cb
    res.raise_for_status()
  File "/home/rm_bot/relman-auto-nag/venv/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://bugzilla.mozilla.org/rest/user?include_fields=_default&names=mnakano.birchill%40mozilla.com
```

### LINKED ISSUE #1677: Employees with outdated Bugzilla account on PMO
This happens when a user changes their account while keeping the link to the old one in PMO. For example, [Atila Butkovits](https://people.mozilla.org/p/p--3h9ia463kmcayophrjx1us) links to [atila.butkovits@softvision.com](https://bugzilla.mozilla.org/user_profile?user_id=661665); however, the active account is [abutkovits@mozilla.com](https://bugzilla.mozilla.org/user_profile?user_id=661920).

```
2022-08-04 23:31:25,571 - ERROR - Tool fuzzing_bisection_without_regressed_by
Traceback (most recent call last):
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/auto_nag/bzcleaner.py", line 746, in run
    self.send_email(date=date)
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/auto_nag/bzcleaner.py", line 679, in send_email
    title, body = self.get_email(date)
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/auto_nag/bzcleaner.py", line 640, in get_email
    data = self.get_email_data(date, bug_ids)
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/auto_nag/bzcleaner.py", line 631, in get_email_data
    bugs = self.get_bugs(date=date, bug_ids=bug_ids)
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/auto_nag/scripts/fuzzing_bisection_without_regressed_by.py", line 250, in get_bugs
    bugs = self.find_regressor_or_needinfo_target(bugs)
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/auto_nag/scripts/fuzzing_bisection_without_regressed_by.py", line 215, in find_regressor_or_needinfo_target
    users = UserActivity(include_fields=["nick"]).get_bz_users_with_status(
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/auto_nag/user_activity.py", line 143, in get_bz_users_with_status
    BugzillaUser(
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/venv/lib/python3.9/site-packages/libmozdata/connection.py", line 134, in wait
    r.result()
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 446, in result
    return self.__get_result()
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/venv/lib/python3.9/site-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/venv/lib/python3.9/site-packages/requests/sessions.py", line 708, in send
    r = dispatch_hook("response", hooks, r, **kwargs)
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/venv/lib/python3.9/site-packages/requests/hooks.py", line 30, in dispatch_hook
    _hook_data = hook(hook_data, **kwargs)
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/venv/lib/python3.9/site-packages/libmozdata/connection.py", line 123, in cb
    res.raise_for_status()
  File "/Users/smujahid/repos/mozilla/relman-auto-nag/venv/lib/python3.9/site-packages/requests/models.py", line 1022, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://bugzilla.mozilla.org/rest/user?include_fields=name&include_fields=can_login&include_fields=last_activity_time&include_fields=last_seen_date&include_fields=nick&names=botond%40mozilla.com&names=jwatt%40jwatt.org&names=hikezoe.birchill%40mozilla.com&names=atila.butkovits%40softvision.com&names=pbone%40mozilla.com&names=nalexander%40mozilla.com&names=m_kato%40ga2.so-net.ne.jp
```

```
{
  "error": true,
  "code": 51,
  "documentation": "https://bmo.readthedocs.io/en/latest/api/",
  "message": "There is no user named 'atila.butkovits@softvision.com'. Either you mis-typed the name or that user has not yet registered for a Bugzilla account."
}
```

_Originally posted by @suhaibmujahid in https://github.com/mozilla/relman-auto-nag/issues/1609#issuecomment-1206010961_

### PR DESCRIPTION
**Pull request by suhaibmujahid**
This fixes #1529 and fixes #1677 for autonag. If the unlinked/new Bugzilla account is based on a Mozilla email, it will be automatically detected.

## Checklist

- [ ] Type annotations added to new functions
- [ ] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/relman-auto-nag/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

File: auto_nag/iam.py [hunk 1]
```diff
@@ -215,42 +215,37 @@ def update_bugzilla_emails(data: Dict[str, dict]) -> None:
...
-            person = users_by_bugzilla_email[bz_user["name"]]
         else:
             raise Exception(f"Can't find {bz_user['name']} in the data")
 
+        if (
+            person.get("found_on_bugzilla")
+            and str(bz_user["id"]) != person["bugzillaID"]
+        ):
+            # If the linked Bugzilla account is still active, we should not
+            # overwrite use the other account.
```
Comment (marco-c) [inline auto_nag/iam.py:246]:
```suggestion
            # overwrite the other account.
```

**Comment by marco-c** [comment]
Isn't #1529 still a problem? If we have a Bugzilla account that isn't linked to a Mozilla email

**Comment by suhaibmujahid** [comment]
> Isn't #1529 still a problem? If we have a Bugzilla account that isn't linked to a Mozilla email

To get moco permissions, the employee must link their Bugzilla account in PMO or use their Mozilla email for their Bugzilla account. After this PR, we will be searching for accounts based on both scenarios. If an account is not linked and is not based on a Mozilla email, the user will not get moco permissions.

**Comment by marco-c** [comment]
OK, so we will cover people who have moco permissions, but the people who don't will still be "uncovered". IDK how much of a problem that is.

**Comment by suhaibmujahid** [comment]
> OK, so we will cover people who have moco permissions, but the people who don't will still be "uncovered". IDK how much of a problem that is.

Mostly it is not a problem for autonag. I filed an issue to follow up on this later.

**Pull request merged**